### PR TITLE
[WIP] [3.0] [Form] fix tests added by #17760 by removing `choices_as_values`

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | not yet |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | - |
- [x] Wait for #17760 being merged in 3.0

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php [hunk 1]
```diff
@@ -94,6 +100,54 @@ public function testExpandedChoicesOptionsTurnIntoChildren()
         $this->assertCount(count($this->choices), $form, 'Each choice should become a new field');
     }
 
+    public function testChoiceListWithScalarValues()
+    {
+        $view = $this->factory->create('Symfony\Component\Form\Extension\Core\Type\ChoiceType', null, array(
+            'choices' => $this->scalarChoices,
+        ))->createView();
```
Comment (iltar) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: If you're merging this into 3.0, you can use `ChoiceType::class`, not sure if that's preferable though.

Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: @iltar, I know, I even proposed to use constants in core type tests (see https://github.com/symfony/symfony/pull/17699#discussion_r51989703) as of 2.8 but this is actually to be consistent with other tests ;)

**Comment by fabpot** [comment]
#17760 merged

**Comment by fabpot** [comment]
#17760 breaks 3.0, @HeahDude Can you have a look at this ASAP please?

**Comment by HeahDude** [comment]
Should be ok. Let me know :)

**Comment by HeahDude** [comment]
Not yet :(

**Comment by HeahDude** [comment]
Ok now I'm waiting for the green lights.

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request closed without merging**